### PR TITLE
Complete the gamecore with interaction functions

### LINKED ISSUE #1: init dev env
1. import https://github.com/SailGame/proto   (use submodule)
2. import gtest/gmock,   grpc (submodule or fetchcontent or whatever it is)
3. set up git action (optional)
4. set up clang-format
5. set up CMakeList skeleton.

### PR DESCRIPTION
**Pull request by MrHate**
Hard to code the relevant test cases :(

### CODE REVIEW

**Review by Gusabary — APPROVED** [review]

**Comment by Gusabary** [comment]
It's better to integrate with gtest/gmock and grpc as soon as possible

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+aux_source_directory(. TEST_DIR)
+aux_source_directory(../src TEST_DIR)
```
Comment (Veiasai) [inline test/CMakeLists.txt]: really don't like this one...
should do add_library(texas_core  ${SOURCE}) in src Cmake.  
And then target_link_libraries(texas_test texas_core)

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+add_executable(texas_test ${TEST_DIR})
+target_link_libraries(texas_test PRIVATE gtest)
+
+include(GoogleTest)
+gtest_add_tests(texas_test "" AUTO)
```
Comment (Veiasai) [inline test/CMakeLists.txt]: yes, this is what I mean.
Then you can use ctest?  or whatever it is.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -1,3 +1 @@
-aux_source_directory(. APP_DIR)
-
-add_executable(texas ${APP_DIR})
+add_library(tcore OBJECT Dummy.cpp Dummy.h TexasScore.cpp TexasScore.h)
```
Comment (Veiasai) [inline src/CMakeLists.txt:1]: add a new line: `target_include_directories(tcore PUBLIC ..)`
And suggest to rename src to tcore.

if you don't rename
use `#include <src/tcore.hpp>` in test/
after rename
`#include <tcore/tcore.hpp>`

File: test/Backdoor.cpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
+#include "Backdoor.h"
+
+#include <iostream>
+
+void DummyBackdoor::DumpDebugMessages(const Dummy &dm, std::ostream &out) {
```
Comment (Veiasai) [inline test/Backdoor.cpp]: optional:  you could overload <<

File: src/Dummy.cpp [hunk 5]
```diff
@@ -9,27 +9,41 @@ const Dummy::uid_t Dummy::Join(const std::string &addr) {
   ++user_count;
   uid2addr.emplace(LastPlayer(), addr);
```
Comment (Veiasai) [inline src/Dummy.cpp:11]: what is addr?

Comment (MrHate) [reply src/Dummy.cpp:11]: The user identity given from client or grpc adapter.

Comment (Veiasai) [reply src/Dummy.cpp:11]: grpc uses unsigned int to mark a user.

File: test/TexasScoreTest.cpp [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+    EXPECT_EQ(ret.royal_straight_flush, 0);
+    EXPECT_EQ(ret.straight_flush, 0);
+    EXPECT_EQ(ret.four_of_a_kind, 0);
+    EXPECT_EQ(ret.full_house, 0);
+    EXPECT_EQ(ret.flush, 0);
+    EXPECT_EQ(ret.straight, 0);
+    EXPECT_EQ(ret.three_of_a_kind, 0);
+    EXPECT_EQ(ret.two_pair, 0);
+    EXPECT_EQ(ret.one_pair, 2);
+    EXPECT_EQ(ret.high_card, 54); // 110110
```
Comment (Veiasai) [inline test/TexasScoreTest.cpp]: could try  0b110110

EXPECT_EQ(ret.high_card, 0b110110);

File: src/Dummy.cpp (hunk 5 above)
Comment (MrHate) [reply src/Dummy.cpp:11]: There will be an adaptor to handle this. Binding with `string` provides extensibility.

**Review by Veiasai — APPROVED** [review]

**Pull request merged**